libfwupd/fwupd-bios-setting.c

```diff
@@ -1020,67 +1001,83 @@ fwupd_bios_setting_to_json(FwupdBiosSetting *self, JsonBuilder *builder)
-
-	str = g_string_new(NULL);
-	fwupd_pad_kv_str(str, FWUPD_RESULT_KEY_NAME, priv->name);
-	fwupd_pad_kv_str(str, FWUPD_RESULT_KEY_BIOS_SETTING_ID, priv->id);
-	fwupd_pad_kv_str(str, FWUPD_RESULT_KEY_DESCRIPTION, priv->description);
-	fwupd_pad_kv_str(str, FWUPD_RESULT_KEY_FILENAME, priv->path);
-	fwupd_pad_kv_int(str, FWUPD_RESULT_KEY_BIOS_SETTING_TYPE, priv->kind);
-	fwupd_pad_kv_str(str, FWUPD_RESULT_KEY_BIOS_SETTING_CURRENT_VALUE, priv->current_value);
-	fwupd_pad_kv_str(str,
-			 FWUPD_RESULT_KEY_BIOS_SETTING_READ_ONLY,
-			 priv->read_only ? "True" : "False");
 
+	fwupd_converter_string_append(str, idt, FWUPD_RESULT_KEY_NAME, priv->name);
+	fwupd_converter_string_append(str, idt, FWUPD_RESULT_KEY_BIOS_SETTING_ID, priv->id);
+	fwupd_converter_string_append(str, idt, FWUPD_RESULT_KEY_DESCRIPTION, priv->description);
+	fwupd_converter_string_append(str, idt, FWUPD_RESULT_KEY_FILENAME, priv->path);
+	fwupd_converter_string_append_int(str, idt, FWUPD_RESULT_KEY_BIOS_SETTING_TYPE, priv->kind);
+	fwupd_converter_string_append(str,
+				      idt,
+				      FWUPD_RESULT_KEY_BIOS_SETTING_CURRENT_VALUE,
+				      priv->current_value);
+	fwupd_converter_string_append(str,
+				      idt,
+				      FWUPD_RESULT_KEY_BIOS_SETTING_READ_ONLY,
+				      priv->read_only ? "True" : "False");
 	if (priv->kind == FWUPD_BIOS_SETTING_KIND_ENUMERATION) {
 		for (guint i = 0; i < priv->possible_values->len; i++) {
 			const gchar *tmp = g_ptr_array_index(priv->possible_values, i);
-			fwupd_pad_kv_str(str, FWUPD_RESULT_KEY_BIOS_SETTING_POSSIBLE_VALUES, tmp);
+			fwupd_converter_string_append(str,
+						      idt,
+						      FWUPD_RESULT_KEY_BIOS_SETTING_POSSIBLE_VALUES,
+						      tmp);
 		}
 	}
 	if (priv->kind == FWUPD_BIOS_SETTING_KIND_INTEGER ||
 	    priv->kind == FWUPD_BIOS_SETTING_KIND_STRING) {
-		fwupd_pad_kv_int(str, FWUPD_RESULT_KEY_BIOS_SETTING_LOWER_BOUND, priv->lower_bound);
-		fwupd_pad_kv_int(str, FWUPD_RESULT_KEY_BIOS_SETTING_UPPER_BOUND, priv->upper_bound);
+		fwupd_converter_string_append_int(str,
+						  idt,
+						  FWUPD_RESULT_KEY_BIOS_SETTING_LOWER_BOUND,
+						  priv->lower_bound);
+		fwupd_converter_string_append_int(str,
+						  idt,
+						  FWUPD_RESULT_KEY_BIOS_SETTING_UPPER_BOUND,
+						  priv->upper_bound);
 		if (priv->kind == FWUPD_BIOS_SETTING_KIND_INTEGER) {
-			fwupd_pad_kv_int(str,
-					 FWUPD_RESULT_KEY_BIOS_SETTING_SCALAR_INCREMENT,
-					 priv->scalar_increment);
+			fwupd_converter_string_append_int(
+			    str,
+			    idt,
+			    FWUPD_RESULT_KEY_BIOS_SETTING_SCALAR_INCREMENT,
+			    priv->scalar_increment);
 		}
 	}
+}
 
-	return g_string_free(str, FALSE);
+/**
+ * fwupd_bios_setting_to_string:
+ * @self: a #FwupdBiosSetting
+ *
+ * Builds a text representation of the object.
+ *
+ * Returns: text, or %NULL for invalid
+ *
+ * Since: 1.8.4
+ **/
+gchar *
+fwupd_bios_setting_to_string(FwupdBiosSetting *self)
+{
+	return fwupd_converter_to_string(FWUPD_CONVERTER(self));
 }
```

Comment:
@superm1 it might be worth removing these too?

Reply:
Yeah drop any dead symbols with our ABI bump, totally agree.

Reply:
Ack, fixed.